pkg/config/virtio.go

```diff
@@ -181,6 +206,109 @@ func (dev *VirtioSerial) FromOptions(options []option) error {
+		return nil, fmt.Errorf("Unknown option for virtio-input devices: %s", inputType)
+	}
+
+	return &VirtioInput{
+		InputType: inputType,
+	}, nil
+}
+
+func (dev *VirtioInput) validate() error {
+	if dev.InputType != VirtioInputPointingDevice && dev.InputType != VirtioInputKeyboardDevice {
+		return fmt.Errorf("Unknown option for virtio-input devices: %s", dev.InputType)
+	} else if dev.InputType == "" {
+		return fmt.Errorf("Device option for virtio-input device not provided")
+	}
+
+	return nil
+}
+
+func (dev *VirtioInput) ToCmdLine() ([]string, error) {
+	if err := dev.validate(); err != nil {
+		return nil, err
+	}
+
+	return []string{"--device", fmt.Sprintf("virtio-input,%s", dev.InputType)}, nil
+}
+
+func (dev *VirtioInput) FromOptions(options []option) error {
+	for _, option := range options {
+		if option.key == VirtioInputPointingDevice || option.key == VirtioInputKeyboardDevice {
+			if option.value != "" {
+				return fmt.Errorf(fmt.Sprintf("Unexpected value for virtio-input %s option: %s", option.key, option.value))
+			}
+			dev.InputType = option.key
+		} else {
+			return fmt.Errorf("Unknown option for virtio-input devices: %s", option.key)
+		}
+	}
+	return dev.validate()
+}
+
+// VirtioGPUNew creates a new gpu device for the virtual machine.
+// The device parameter is the virtio-gpu device that will be added to the machine
+func VirtioGPUNew(device string) (VirtioDevice, error) {
+	if device != VirtioGPUDisplayDevice {
+		return nil, fmt.Errorf("Unknown option for virtio-gpu devices: %s", device)
+	}
+
+	return &VirtioGPU{
+		Device:  device,
+		UsesGUI: false,
+	}, nil
+}
+
+// VirtioGPUNewWithGUI creates a new gpu device for the virtual machine with the
+// graphical user interface enabled. The device parameter is the virtio-gpu
+// device that will be added to the machine
+func VirtioGPUNewWithGUI(device string) (VirtioDevice, error) {
+	dev, err := VirtioGPUNew(device)
+	if err != nil {
+		return nil, err
+	}
+	dev.(*VirtioGPU).UsesGUI = true
+	return dev, nil
+}
+
+func (dev *VirtioGPU) validate() error {
+	if dev.Device != VirtioGPUDisplayDevice {
+		return fmt.Errorf("Unknown option for virtio-gpu devices: %s", dev.Device)
+	} else if dev.Device == "" {
+		return fmt.Errorf("Device option for virtio-gpu device not provided")
+	}
+
+	return nil
+}
+
+func (dev *VirtioGPU) ToCmdLine() ([]string, error) {
+	if err := dev.validate(); err != nil {
+		return nil, err
+	}
+
+	return []string{"--device", fmt.Sprintf("virtio-gpu,%s", dev.Device)}, nil
+}
+
+func (dev *VirtioGPU) FromOptions(options []option) error {
+	for _, option := range options {
+		if option.key == VirtioGPUDisplayDevice {
+			if option.value != "" {
+				return fmt.Errorf(fmt.Sprintf("Unexpected value for virtio-gpu %s option: %s", option.key, option.value))
+			}
+			dev.Device = option.key
+		} else {
+			return fmt.Errorf("Unknown option for virtio-gpu devices: %s", option.key)
```

Comment:
consider moving this "up" as an if-condition and inverting the condition.   if option.key != VirtGPUDisplaDevice ...